Drop-In/src/androidTest/java/com/braintreepayments/api/CardDetailsFragmentUITest.kt

```diff
@@ -0,0 +1,380 @@
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        onView(withId(R.id.bt_card_form_expiration)).check(matches(isFocused()))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_displaysCVVAndPostalCodeWhenPresentInConfiguration() {
+        val args = Bundle()
+        args.putParcelable("EXTRA_DROP_IN_REQUEST", DropInRequest().clientToken(Fixtures.CLIENT_TOKEN))
+        args.putParcelable("EXTRA_CARD_FORM_CONFIGURATION", CardFormConfiguration(true, true))
+        args.putString("EXTRA_CARD_NUMBER", VISA)
+        val scenario = FragmentScenario.launchInContainer(CardDetailsFragment::class.java, args, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        onView(isRoot()).perform(waitFor(500))
+
+        onView(withId(R.id.bt_card_form_cvv)).check(matches(isDisplayed()))
+        onView(withId(R.id.bt_card_form_postal_code)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_hidesCVVandPostalCodeWhenNotPresentInConfiguration() {
+        val args = Bundle()
+        args.putParcelable("EXTRA_DROP_IN_REQUEST", DropInRequest().clientToken(Fixtures.CLIENT_TOKEN))
+        args.putParcelable("EXTRA_CARD_FORM_CONFIGURATION", CardFormConfiguration(false, false))
+        args.putString("EXTRA_CARD_NUMBER", VISA)
+        val scenario = FragmentScenario.launchInContainer(CardDetailsFragment::class.java, args, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        onView(isRoot()).perform(waitFor(500))
+
+        onView(withId(R.id.bt_card_form_cvv)).check(matches(not(isDisplayed())))
+        onView(withId(R.id.bt_card_form_postal_code)).check(matches(not(isDisplayed())))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_withDropinRequestRequiringCardholderName_setsCardFormsCardholderNameToRequired() {
+        val args = Bundle()
+        args.putParcelable("EXTRA_DROP_IN_REQUEST", DropInRequest().clientToken(Fixtures.CLIENT_TOKEN).cardholderNameStatus(FIELD_REQUIRED))
+        args.putParcelable("EXTRA_CARD_FORM_CONFIGURATION", CardFormConfiguration(false, false))
+        args.putString("EXTRA_CARD_NUMBER", VISA)
+        val scenario = FragmentScenario.launchInContainer(CardDetailsFragment::class.java, args, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        onView(isRoot()).perform(waitFor(500))
+
+        onView(withId(R.id.bt_card_form_expiration)).perform(typeText(ExpirationDate.VALID_EXPIRATION))
+        scenario.onFragment { fragment ->
+            assertFalse(fragment.cardForm.isValid)
+        }
+
+        onView(withId(R.id.bt_card_form_cardholder_name)).perform(typeText("Brian Tree"))
+        scenario.onFragment { fragment ->
+            assertTrue(fragment.cardForm.isValid)
+        }
+    }
+
+    @Test
+    fun whenStateIsRESUMED_withoutSaveCardCheckBoxPropertiesOnDropInRequest_saveCardCheckboxCheckedAndHidden() {
```

Comment:
Nitpick - but test names could use more consistent format. It isn't a huge deal, but I think makes it easier to read through the tests.

Example:

```
whenStateIsRESUMED_whenDropInRequestRequiresCardholderName_... 
whenStateIsRESUMED_whenDropInRequestUsesDefaultSaveCardCheckboxProperties_...
```

Reply:
Yeah I agree. We ported most of the names from the original tests. They could use a second look.